plugins/meta/bandwidth/main.go

```diff
@@ -135,9 +154,20 @@ func cmdAdd(args *skel.CmdArgs) error {
 		return err
 	}
 
-	//no traffic shaping was requested, so just no-op and quit
+	var bandwidth *BandWidthEntry
 	if conf.IngressRate == 0 && conf.IngressBurst == 0 && conf.EgressRate == 0 && conf.EgressBurst == 0 {
-		return types.PrintResult(conf.PrevResult, conf.CNIVersion)
+		bandwidth = conf.RuntimeConfig.BandWidth
+		//no traffic shaping was requested, so just no-op and quit
+		if bandwidth == nil || (bandwidth.IngressRate == 0 && bandwidth.IngressBurst == 0 && bandwidth.EgressRate == 0 && bandwidth.EgressBurst == 0) {
```

Comment:
I suggest you add an `IsZero` method on the BandwidthEntry type - makes these comparisons simpler.

Reply:
Also, if you make the nested BandwidthEntry struct in config a pointer, this can be simpler.